src/preact/base-element.js

```diff
@@ -1128,7 +1110,7 @@ function matchChild(element, defs) {
     const def = defs[match];
     const selector = typeof def == 'string' ? def : def.selector;
     if (matches(element, selector)) {
-      return def;
+      return typeof def == 'string' ? def : {name: match, ...def};
```

Comment:
The `...def` is very iffy for typedef structs. It probably breaks the overall type. I realize why it's here. But maybe we should just return a `match` and let the caller do the `def = props[match]`?

Reply:
Good idea, done.